### PR TITLE
feat: set default builderBoostFactor to 90

### PR DESCRIPTION
**Pull request by fredrik0x**
**Motivation**

Improve censorship resistance

**Description**

In order to help increase censorship resistance, I propose to change the default builderBoostFactor to 90. This means validators will prioritize local block building unless the bid from the external block builder is 10% or higher than what the validator would receive when building locally.

Looking at ([stats](https://censorship.pics/)), it can be seen that currently 63.7% of external builders are censoring transactions compared to 8.53% of validators who do local block building, so setting a minimum 10% as default can help increase the overall censorship resistance of the network.

It is still easy for users to opt out of this by manually setting the flag to 100, but many are likely to use the default which could help with censorship resistance for the network.

### CODE REVIEW

**Comment by fredrik0x** [comment]
Nimbus: https://github.com/status-im/nimbus-eth2/pull/6103
Lighthouse: https://github.com/sigp/lighthouse/pull/5441
Lodestar: https://github.com/ChainSafe/lodestar/pull/6568
Teku: https://github.com/Consensys/teku/pull/8108
Prysm: https://github.com/prysmaticlabs/prysm/pull/13772

**Comment by nflaig** [comment]
> This means validators will prioritize local block building unless the bid from the external block builder is 10% or higher than what the validator would receive when building locally.

Setting a builder boost factor of 90 means the builder block value will be reduced by 10% but this is not the same as the increasing the local block value by 10%.

Let's assume a local block value of 100, if builder block has value of 110 (10% more) it will still pick local block
> 110 * 0.90 = 99 < 100

We have [notes about the calculation](https://chainsafe.github.io/lodestar/validator-management/vc-configuration/#calculating-builder-boost-factor-with-examples) in our docs.

I don't think it matters in this case as the difference is marginal.

**Review by ensi321 — DISMISSED** [review]
Looks good to me. Would prefer holding off merging until ACD and see if testing is needed

**Comment by philknows** [comment]
We should do a follow up PR on this to make users aware of the default setting on our documentation.

**Comment by philknows** [comment]
I have a question to this after giving it some more thought about how we deal with `builder.selection` UX. If I'm not mistaken, we need to have `--builder.selection maxprofit` in order to benefit from `--builderBoostFactor X` ? 

This will essentially break the idea of having a `maxprofit` setting because if the default value is set to anything other than 100, it's really not `maxprofit` anymore. I would hate breaking this, but maybe we need an additional setting that is the actual default like `default` or `boostdefault` or something better. I'm just concerned that `maxprofit` is no longer accurately defining what it's doing and is misleading to users, even if it was disclaimed in the docs. We could also consider just having a default setting and getting rid of the `maxprofit` altogether.

**Comment by nflaig** [comment]
@philknows that's a good point, maybe we add another alias that translates to boost factor 90?

Also on another thought, this change might be good for censorship resistance but it might also hurt the credible neutrality as something like this should be enforced via market forces or changes that do not reduce profits of stakers without them potentially being aware of it, not everybody closely reads release notes or follow the news.

So the question for me is also should the client software be opinionated about something like this because "we know what's best for the user (and the network)" or should the user decide themselves.

Might be better to push this similar to client diversity

**Comment by philknows** [comment]
> So the question for me is also should the client software be opinionated about something like this because "we know what's best for the user (and the network)" or should the user decide themselves.

To be fair, it'll be disclosed and the user has full ability to change their builder selection preference at free will. It's just the name `maxprofit` that is misleading if 90 is actually the default. I think `--builder.selection default` is ambiguous enough that if someone wanted to understand what "default" means, they would ideally look at the docs to understand what that means or what that number is. Our default is a suggestion based on our analysis of what is safe for the network (which is credibly neutral imo), but ultimately users can configure how they want, which is is the more important part.

**Comment by nflaig** [comment]
> our analysis of what is safe for the network 

I don't think this analysis is profound at all. Censoring transactions is economically irrational and if more contracts (like Uniswap) would be on the OFAC list builders in US would be forced to relocate to a different jurisdiction or would become less competitive and lose their market share sooner or later.

Websites like censorship.pics claiming that a certain % of slots are being censored is also not correct, looking at [Tornado cash router](https://etherscan.io/address/0xd90e2f925DA726b50C4Ed8D0Fb90Ad053324F31b) it doesn't even come close to a tx every slot. And if there is no tx like this in the mempool that the builder explicitly does not include then claiming the slot is being censored is just not correct.

So how exactly do we explain to the user that it was in the interest of the network to produce the block locally and missing out on some rewards? Maybe I am too much of a capitalist here but generally I think "state intervention" should be avoided if it's not critical.

**Comment by fredrik0x** [comment]
My thinking is that people have always needed to change default settings in order to not build local blocks 100% of the time (e.g. they have to explicitly specify a relay or mev-boost proxy), so max profit has never been the default setting for a validator. When they are viewing the documentation on how to configure lodestar to use an external builder, they would also see the default builder boost and how to adjust it.

I am primarily looking at this from the angle of making Ethereum more censorship resistant, but from a capitalist point of view I can also see some positive long term effects for validators (and the whole ecosystem). EIP-1559 was for example seen by many miners as short term negative as the basefee was lost, but the long term effects for the ecosystem appears to have been positive. With this proposed change, builders would need to submit blocks that are at least 10% more profitable than the local mempool, which - assuming builders have or can achieve that profit margin - could mean validators end up receiving higher rewards. Long term I also believe increased censorship resistance will have a positive effect on Ethereum.

File: docs/pages/validator-management/vc-configuration.md
Comment (nflaig) [inline docs/pages/validator-management/vc-configuration.md]: The docs are a bit misleading here, `maxprofit` is actually not the default but `executiononly` is. We only produce builder blocks if `--builder` flag is set or if `--builder.selection` is set to `maxprofit | builderalways | builderonly`, see [builder.selection docs](https://chainsafe.github.io/lodestar/validator-management/validator-cli/#-builderselection)

**Comment by nflaig** [comment]
> so max profit has never been the default setting for a validator

Our docs are bit misleading here, see https://github.com/ChainSafe/lodestar/pull/6568#discussion_r1536776107. It is not actually the default setting.

> I am primarily looking at this from the angle of making Ethereum more censorship resistant

I appreciate the push for this and at first I also thought it is a good idea but after thinking about this a bit more it does not seem reasonable to change this right now. Censorship resistance is a really vague term imo, it hard to define when is Ethereum is or is not censorship resistant. But it is easy to tell if Ethereum is censoring or not, and looking at tornado cash txs, those are clearly not censored.

Imo what makes Ethereum censorship resistant is a large set of independent node operators which is the case for Ethereum. And I have high confidence in those operators (especially solo stakers) to do the right thing if there was meaningful / measurable censorship on Ethereum.

The change proposed here might increase the % of locally build blocks slightly and make the graphs on censorship.pics and other similar websites look prettier but other than that doesn't meaningfully change anything. So it seems more like a marketing measure than solving a real issue.

And don't get me wrong, building local blocks is great for many reasons, like not extracting value from user transactions which is also why I use a high min-bid myself. But I don't think right now it matters for censorship resistance.

**Comment by philknows** [comment]
Just wanted to include this reference with data for some discussion about whether or not this is a meaningful change, which would require a bit of re-working on how to handle the UX.

https://hackmd.io/@dataalways/resilience

**Comment by nflaig** [comment]
> data for some discussion about whether or not this is a meaningful change

We also need to keep in mind that this is highly dependent on market conditions, in a bull market with high price volatility, the local block needs to be boosted much more as MEV increases disproportional compared to local block value.

Considering this, educating users about this setting as an alternative to `min-bid` seem to be the better way to approach this.

**Comment by philknows** [comment]
Based on our standup on Apr 2, we decided that we will move forward with this with [feedback from the community](https://twitter.com/lodestar_eth/status/1772679499928191044) and with [additional data that shows a very mild difference](https://github.com/ChainSafe/lodestar/issues/6602#issuecomment-2025108408) for inclusion. Although this will not end the conversation on censorship resistance, there is harder support for https://github.com/ChainSafe/lodestar/issues/6602 which data points will make more of a difference.

Once we figure out the alias option name because it's no longer `maxprofit` we can include this. Some options thrown in was `recommended` or `default` as an alias for 90. PR is currently set with `default` as the alias and if there's no additional comments to the alias, we can merge.

**Review by philknows — DISMISSED** [review]

File: packages/cli/src/util/proposerConfig.ts
Comment (nflaig) [inline packages/cli/src/util/proposerConfig.ts:110]: this is a breaking change, users who have set `--builder.selection=maxprofit` will now get an error
```
Error: Invalid input for builder selection, check help
```

**Review by nflaig — CHANGES_REQUESTED** [review]
I think we should not change the `maxprofit` behavior and keep it as is. Instead I think it's better to add a new selection option and use that as default instead of `maxprofit`. This would ensure that users who have set this explicitly won't be affected and it's no longer a breaking change.

File: packages/beacon-node/src/api/impl/validator/index.ts
Comment (nflaig) [inline packages/beacon-node/src/api/impl/validator/index.ts]: I reverted the changes on the beacon node side as in case of Lodestar the boost factor is dictated by the validator client and setting a different default here is undocumented behavior we should avoid.

File: packages/cli/src/cmds/validator/options.ts
Comment (nflaig) [inline packages/cli/src/cmds/validator/options.ts]: Since we have the `default` for users that do not change the defaults it makes more sense to only apply the manually set boost factor in case of `maxprofit` as it is anyhow a setting for more advanced users.

File: packages/cli/test/unit/validator/proposerConfigs/validData.yaml
Comment (nflaig) [inline packages/cli/test/unit/validator/proposerConfigs/validData.yaml]: Might encourage users to set it even lower :)

**Review by nflaig — DISMISSED** [review]

**Review by nflaig — DISMISSED** [review]

**Comment by philknows** [comment]
Got a conflict here before merging.

**Review by nflaig — APPROVED** [review]
Resolved merge conflicts, and are we ok with the semantics around this, i.e. boost factor can only be configured if `maxprofit`, and `default` uses the "default" boost factor we set?

**Review by g11tech — COMMENTED** [review]
lgtm

**Comment by wemeetagain** [comment]
:tada: This PR is included in [v1.18.0](https://github.com/ChainSafe/lodestar/releases/tag/v1.18.0) :tada:

**Pull request merged**
